.github/workflows/versioning.yml

```diff
@@ -0,0 +1,50 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This workflow creates a new tag following the guidelines in the README.
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'mock-responses**'
+
+jobs:
+  add-tag:
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Get current version number
+        run: |
+          [[ $(git tag | tail -n1) =~ ^v([0-9]+)\.([0-9]+)$ ]]
+          echo "MAJOR=${BASH_REMATCH[1]}" >> $GITHUB_ENV
+          echo "MINOR=${BASH_REMATCH[2]}" >> $GITHUB_ENV
+      - name: Find new version number
+        run: |
+          # If any change to mock-responses other than a new file added
```

Comment:
This workflow should only run with changes to `mock-responses` due to:
```
    paths:
      - 'mock-responses**'
```
Checking if files were added with `A` means we'll still have to check with `a` because files being added does not imply that files weren't modified, right?

Reply:
If a file have been added, there's no need to check if one has been modified, as it would be major version bump. If no file was added, but there were affected files in the `mock-responses` directory, then is minor.

Is there a case missing?